### PR TITLE
[processing] porting some small fixes to master

### PR DESCRIPTION
**Pull request by ghtmtt**
This PR ports some fixes from master_2 to master. PR of master_2 are: #3544, #3612, #3616 

Fixes are very small and should not create any conflicts.

### CODE REVIEW

File: python/plugins/processing/gui/ScriptEditorDialog.py [hunk 1]
```diff
@@ -139,19 +138,10 @@ def __init__(self, algType, alg):
...
-
-    def decreaseFontSize(self):
-        font = self.editor.defaultFont
-        self.editor.setFonts(font.pointSize() - 1)
-        self.editor.initLexer()
 
     def showSnippets(self, evt):
         popupmenu = QMenu()
-        for name, snippet in list(self.snippets.items()):
+        for name, snippet in self.snippets.iteritems():
```
Comment (m-kuhn) [inline python/plugins/processing/gui/ScriptEditorDialog.py]: In python 2 it's `iteritems()` in python 3 it's `items()` and for compatibility `list(items())` was there.

File: python/plugins/processing/gui/ScriptEditorDialog.py [hunk 2]
```diff
@@ -173,9 +163,9 @@ def closeEvent(self, evt):
     def editHelp(self):
         if self.alg is None:
             if self.algType == self.SCRIPT_PYTHON:
-                alg = ScriptAlgorithm(None, str(self.editor.text()))
+                alg = ScriptAlgorithm(None, unicode(self.editor.text()))
```
Comment (m-kuhn) [inline python/plugins/processing/gui/ScriptEditorDialog.py]: Nope, python3 doesn't have unicode. It has str. But better remove it altogether, there's no reason to cast a str to an str.

Comment (m-kuhn) [inline python/plugins/processing/gui/ScriptEditorDialog.py]: (Applies to all of the str->unicode items in this pull request as far as I can tell)

**Review by m-kuhn — CHANGES_REQUESTED** [review]
master should be python 3 only.
master_2 should be python 2 / python 3 compatible (or if in doubt prefer python 2)

**Review by m-kuhn — APPROVED** [review]
Looks good to me. I think the filename and items() wouldn't need any explicit conversion but at least it shouldn't hurt in the current state.
Thanks Matteo

**Comment by ghtmtt** [comment]
I just wanted to add my small changes, so sorry if I'm not fast and reliable with other changes ;) 
Thanks Matthias for the review and the help!

**Comment by m-kuhn** [comment]
Good stuff, please continue.
Just waiting for the travis feedback now.

**Pull request merged**